### DISCUSSION TITLE
Folder structure

### DISCUSSION THREAD

**Discussion by EstherPlomp** [EstherPlomp/TNW-RDM-101#10] [General]
Based on the [CookieCutter assignment](https://estherplomp.github.io/TNW-RDM-101/08-Realising-FAIR-Organisation.html#assignment-using-cookiecutter), or manual folder creation, please share your folder structure below:

**Comment by RamonvanValderen** [comment]
As general template I'll use the following folder structure:

├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── requirements.txt
├── 1-bin <- Compiled and external code, ignored by git (PG)
│ └── external <- Any external source code, ignored by git (RO)
├── 2-config <- Configuration files (HW)
├── 3-simulation_input <- All input files to run a simulation, software specific
│ └── metadata <- All metadata regarding simulation
├── 4-simulation_output <- All raw data generated from simulation
├── 5-data_analysis_methods <- All files related to the analysis of the data, such as Python, Julia or Matlab files
├── 6-docs <- Documentation notebook for users (HW)
│ ├── manuscript <- Manuscript source, e.g., LaTeX, Markdown, etc. (HW)
│   └── reports <- Other project reports and notebooks (e.g. Jupyter, .Rmd) (HW)
├── 7-results
│ ├── figures <- Figures for the manuscript or reports (PG)
│   └── output <- Other output for the manuscript or reports (PG)
└── src <- Source code for this project (HW)

I've created my own Cookiecutter template and posted it on my Github account incase anyone is interested in using this as well: https://github.com/RamonvanValderen/cookiecutter-simulations. This template is setup specifically for projects where data is generated using multiphysics simulations (like computational fluid dynamics), which is subsequently post-processed and analyzed.

**Reply by EstherPlomp** [reply]
Great that you posted your cookiecutter template on GitHub! I think this looks very clear, so no comments from my side 👍

**Comment by TamaraJankovic03** [comment]
This will be mine folder structure:

├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── requirements.txt
├── 1-bin <- Compiled and external code, ignored by git (PG)
│ └── external <- Any external source code, ignored by git (RO)
├── 2-config <- Configuration files (HW)
├── 3-simulation_input <- All input data required to start simulation (RO)
│ └── in_process_sequence_1
│ └── in_process_sequence_2
│ └── in_process_sequence_3
├── 4-simulation_output <- All raw data generated from simulation (PG)
│ └── out_process_sequence_1
│ └── out_process_sequence_2
│ └── out_process_sequence_2
├── 5-simulation_analysis <- All data generated from simulation analysis (HW)
│ └── analysis_process_sequence_1
│ └── analysis_process_sequence_2
│ └── analysis_process_sequence_1
├── 6-docs <- Documentation notebook for users (HW)
│ ├── manuscript <- Manuscript source, e.g., LaTeX, Markdown, etc. (HW)
│ └── reports <- Other project reports and notebooks (e.g. Jupyter, .Rmd) (HW)
├── 7-results
│ ├── figures <- Figures for the manuscript or reports (PG)
│ └── output <- Other output for the manuscript or reports (PG)
└── src <- Source code for this project (HW)

**Reply by EstherPlomp** [reply]
Looks very clear @TamaraJankovic03 - well done!

**Comment by bertvanherck** [comment]
├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── requirements.txt
├── 1-bin <- Compiled and external code, ignored by git (PG)
│ └── external <- Any external source code, ignored by git (RO)
├── 2-config <- Configuration files (HW)
├── 3-raw_images <- All unprocessed microscopy images
│ └── metadata <- All metadata regarding the images
├── 4-image_analysis_output <- All raw data generated by the analysis
├── 5-image_analysis_methods <- All files related to the analysis of the images, such as Python files
├── 6-docs <- Documentation notebook for users (HW)
│ ├── manuscript <- Manuscript source, e.g., LaTeX (HW)
│ └── reports <- Other project reports and notebooks (e.g. Jupyter) (HW)
├── 7-results
│ ├── figures <- Figures for the manuscript or reports (PG)
│ └── output <- Other output for the manuscript or reports (PG)
└── src <- Source code for this project (HW)

**Reply by EstherPlomp** [reply]
Looks very clear @bertvanherck - No comments from my side 👍

**Comment by SDev5** [comment]
I will try to implement the following folder structure: 

├── .gitignore 
├── CITATION.md 
├── LICENSE.md 
├── README.md 
├── requirements.txt 
├── bin <- Compiled and external code, ignored by git (PG)  
│ └── external <- Any external source code, ignored by git (RO) 
├── config <- Configuration files (HW) 
├── Theory model <- Files for simulation and analysis of theoretical model (HW)
 | ├── Simulation output <- Raw output data of simulations (PG)
 | └── Analysis output <- Output of analysis of simulation data (PG)
├── Device design <- Files for design and simulation of circuit design (HW)
 |└── Electromagnetic simulation output <- Output of electromagnetic simulations of circuit design (PG)
├── Fabrication data <- Data collected during fabrication of device (RO)
 | ├── Fabrication step 1 
 | ├── Fabrication step 2 
 | └── Etc. 
├── Measurement data <- Data collected during measurement of device (RO)
│├── Measurement type 1
│├── Measurement type 2
| └── Etc. 
├── Measurement data analysis <- Files for the analysis of the measurement data (HW)
├── docs <- Documentation notebook for users (HW) 
│ ├── manuscript <- Manuscript source, e.g., LaTeX, Markdown, etc. (HW) 
│ └── reports <- Other project reports and notebooks (e.g. Jupyter, .Rmd) (HW) 
├── results 
│ ├── figures <- Figures for the manuscript or reports (PG) 
│ └── output <- Other output for the manuscript or reports (PG)

**Reply by EstherPlomp** [reply]
Looks great! Just a note on spaces in folder names: this can cause complications if you're using scripts; not every programming language responds well to spaces in file/folder names (although there are work-arounds).

**Comment by MBeatrizDiniz** [comment]
I will try to implement the following folder structure.
[Folder structure.txt](https://github.com/EstherPlomp/TNW-RDM-101/files/10951183/Folder.structure.txt)

**Reply by EstherPlomp** [reply]
Looks great! Just a note on using spaces in folder (or file) names: not every programming language is able to handle this well. It can be easier to include _ or - in file/folder names. 

I see you're using R? RMarkdown (and now also [Quarto](https://quarto.org/)) also has some nice options for setting up reports.

**Comment by begum94** [comment]
I checked some examples from github but I thought that the structure that I am using is consistent enough for my research and I wanted to share it. Herein, I reported the main structure with more detail on how I store the experimental data. I am carrying out an experimental research hence I do not have any coding script for now.
├── tu delft

 ├── 1. general
          └── 1.1 courses
          └── 1.2 finance
          └── 1.3 registration/introductive documents

 ├── 2. research
         └── 2.1 drafts <- any draft for conference papers and articles (.docx) with related graphics (.ai)
         └── 2.2 experiments  
└── 2.2.1 YYMMDD - instrument/method - exp name/material <- exp folder indicating the date, instrument and material
└── 2.2.1.1 raw data <-.csv data
└── 2.2.1.2 YYMMDD - report - exp name <-a report containing author, experimental details and outputs (.docx)
└── 2.2.1.3 YYMMDD - processed data - exp name <-.xls or .opju document including processed data and plots
└── 2.2.2 etc.
└── 2.3 literature <-in sync with mendeley (.pdf)
└── 2.4 safety <-safety data sheets (.docx)

   ├── 3. shared
         └── 3.1 analytical data <-folders in experiment section distributed based on the instrument that was used and also shared with group
         └── 3.2 lab journal <-a summary of my hard copy lab journal shared with group. Each experiment linked to related experiment folder

   ├── 4. TA <-teaching assistant folder
         └── 4.1 LO2

I think my reports in experiment folder can be considered as readme file. However, my hard copy lab journal is quite important to follow raw data. My digital lab journal does not contain that much of detail (takes too much time). I am thinking that I can scan my lab journal and add the file to my stored data. Do you have other suggestions?

**Reply by EstherPlomp** [reply]
Looks very clear! 

Scanning your lab journal is a good practice in any case, because it will ensure you're following the 3-2-1 back up strategy. If you only have your physical copy and something happens to it - all the information will be lost!

I don't think I have any suggestions on how to speed this up - if anyone does, feel free to add your comments!

**Comment by MiriamCammaert** [comment]
I am planning to separate my data between two folders. I will set-up a project Drive (U:) for my read-only (raw) data and 
for documentation on this data. Besides the U: Drive, I will also use the M: Drive that will also include all "personal" files 
for my PhD including Graduate School course documents, literature, and so on. 

Structure of the U: Drive folder: 
.
├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── requirements.txt
├── bin                <- Compiled and external code, ignored by git (PG)
│   └── external       <- Any external source code, ignored by git (RO)
├── config             <- Configuration files (HW)
├── Research line 1 (Physiology)    <- All project data on research line 1, ignored by git
│   ├── processed      <- The final, canonical data sets for modeling. (PG)
│   ├── raw            <- The original, immutable data dump. (RO)
│   └── metadata       <- All metadata regarding research line 1
│   ├── figures        <- Figures for the manuscript or reports (PG)
│   └── output         <- Other output for the manuscript or reports (PG)
├── Research line 2 (Metabolic Engineering)    <- All project data on research line 1, ignored by git
│   ├── processed      <- The final, canonical data sets for modeling. (PG)
│   ├── raw            <- The original, immutable data dump. (RO)
│   └── metadata       <- All metadata regarding research line 2
│   ├── figures        <- Figures for the manuscript or reports (PG)
│   └── output         <- Other output for the manuscript or reports (PG)
└── src                <- Source code for this project (HW)

**Reply by EstherPlomp** [reply]
I like the distinguishment between the different research lines as a way to separate directories! 

Small note that using spaces in folder/directory names may complicate things when working in the command line or when working with scripts, in that case it is easier to name directories with - or _ instead of spaces (Research-line_1).

**Comment by maverheijen** [comment]
I will make separate folders for each research line within my PhD, for which I will try to implement the following structure:

├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── requirements.txt
├── 1 - bin                <- Compiled and external code, ignored by git _(PG)_
│   └── external        <- Any external source code, ignored by git _(RO)_
├── 2 - research design                          <- 'Configuration' files _(HW)_
│   ├── 2.1 - experimental set-up           <- Files related to reactor setup (parameters) and lab protocols. _(HW)_
│   └── 2.2 - model config                      <- Configuration files (.py, .csv) _ (HW)_
├── 3 - experimental data         <- All project data, ignored by git
│   ├── 3.1 - raw                        <- The original, immutable data of reactor measurements and analysis calibration lines (.csv) . _(RO/HW?)_
│   └── 3.2 - temp                     <- Intermediate data that has been transformed. (.csv)  _(PG)_
│   ├── 3.3 - processed             <- The final, canonical data sets for modeling: analysis calibration lines and the processed reactor data with these calibration lines. _(PG)_
├── 4 - model output               <- All project data, ignored by git
│   ├── 4.1 - raw                        <- The original, immutable data from the model. _(RO)_
│   │   ├── 4.1.1. - yyyy-mm-dd hh_mm h _summary of settings_                     
│   │   ├── etc
│   └── 4.2 - temp                     <- Intermediate data that has been transformed. _(PG)_
│   │   ├── 4.2.1. - yyyy-mm-dd hh_mm h _summary of settings_                     
│   │   ├── etc
│   ├── 4.3 - processed             <- The final, canonical merged data sets from different sets of raw data. _(PG)_
├── 5 - results
│   ├── figures                          <- Figures for the manuscript or reports (PG)
│   └── output                          <- Other output for the manuscript or reports (PG)
├── 6 - docs                             <- Documentation notebook for users _(HW)_
│   ├── manuscript                   <- Manuscript source, e.g., LaTeX, Markdown, etc. _(HW)_
│   └── reports                          <- Other project reports and notebooks (e.g. Jupyter, .Rmd) _(HW)_
├── 7 - presentations               <- Rough overviews of results for communication in meetings _(HW)_
└── src                                      <- Source code for this project (HW)

Other folders for my PhD relating to finance/graduate school/literature are not included here, but they also are not necessary to share.

**Reply by EstherPlomp** [reply]
Looks very clear! 

In general it is good practice to keep at least one copy of the raw data in read only format so that you don't accidentality make any irreversible changes. Perhaps for 3.1 you could split it up into a raw read only and a raw 'editable' directory?

Small note that using spaces in folder/directory names may complicate things when working in the command line or when working with scripts, in that case it is easier to name directories with - or _ instead of spaces (model-output).

**Comment by rooswouters** [comment]
I will make a seperate folder for each research line in my PhD, as such I have indicated the general structure of my current workfolder below. For the raw data I have a seperate storage location which largely maintains the same structure as indicated under Research;

├── tu delft

├── 00. Misc

├── 01. Literature
└─── All raw literature (without annotations) (.pdf) <- in sync with Mendeley

├── 02. Research
└─── 02.00 Protocols
└──── YYYYMMDD Technique/Material 

└─── 02.01 Research Line 1
└──── 00000000 READ ME <- Aim; Approach; Results (short)

└──── 02.01.01 CloneManager <- Theoretical Construction Data
└───── READ ME <- Aim; Approach; Results (detailed)
└───── Plasmids <- Completed Constructs
└───── Parts <- Parts required to make constructs


└──── 02.01.02 Sequencing <- Verification Construction Data
└───── YYYYMMDD Construct - Sequencing Technique - Name Raw Data Sample
└────── READ ME <- Aim; Approach; Results (detailed) 
└────── Raw Data (.xlxs)
└────── Alignment File (.cm5)
└────── Figures (.ppx, graphpad)


└──── 02.01.0X Other Characterization Techniques <- e.g., Enzyme Assay, Growth Study
└───── YYYYMMDD Strain(s) - Condition (if relevant)
└────── READ ME <- Aim; Approach; Results (detailed) 
└────── Raw Data (.xlxs)
└────── Figures(.ppx, graphpad)



├── 03. Teaching
└─── YY-YY Course 
└──── Lectures


├── 04. Graduate School
└─── Courses
└──── XX.XX Course Name
└───── Lectures
└───── Assignments
└───── etc.
└─── Certificates
└─── Administration
└──── PhD Agreement

├── 05. Meetings
└─── YYYYMMDD Subject_Attendees
└──── Notes


├── 06. Manuscripts
└─── Thesis
└──── Chapter 1
└───── YYYYMMDD Chapter 1 vX.X
└─── Other Manuscripts
└──── YYYYYMMDD Title

**Reply by EstherPlomp** [reply]
Looks great! I like the inclusion of the teaching, graduate school and meeting information. Also the manuscripts folder/directory is very handy!

**Comment by sweerdenburg1** [comment]
This is the file organization of my project. In the data folder I store my digital lab journal and I have a subfolder where I store all my analysis scripts (.py files) and a subfolder for each measurement technique that I use (e.g. UV-Vis, TEM, ATR-FTIR, etc.). Inside each measurement technique subfolder, there is a subfolder for each experiment tag, which corresponds to an experiment documented in the lab journal. Inside the experiment subfolder is the raw data and a separate folder containing the processed data.
├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── requirements.txt
├── bin                <- Compiled and external code, ignored by git (PG)
│   └── external       <- Any external source code, ignored by git (RO)
├── config             <- Configuration files (HW)
├── data               <- All project data, ignored by git
│   ├── lab_journal.odt
│   ├── scripts
│   ├── measurement technique
│   │   └── experiment tag
│   │   │   └── raw            <- The original, immutable data dump. (RO)
│   │   │   └── processed      <- The final, canonical data sets for modeling. (PG)
├── docs               <- Documentation notebook for users (HW)
│   ├── manuscript     <- Manuscript source, e.g., LaTeX, Markdown, etc. (HW)
│   └── reports        <- Other project reports and notebooks (e.g. Jupyter, .Rmd) (HW)
├── results
│   ├── figures        <- Figures for the manuscript or reports (PG)
│   └── output         <- Other output for the manuscript or reports (PG)
└── src                <- Source code for this project (HW)

**Reply by EstherPlomp** [reply]
Looks very clear! 

Small note that using spaces in folder/directory names (measurement technique / experiment tag) may complicate things when working in the command line or when working with scripts, in that case it is easier to name directories with - or _ instead of spaces. Which you might already do because I see 'lab_journal.odt' and you're probably just using these names as an example, but I'll point it out to be sure :)

**Comment by MarijnVDBr** [comment]
[README.md](https://github.com/EstherPlomp/TNW-RDM-101/files/10957696/README.md)

An example for 3 types of experimentally collected data: For each type of instrument, I collect data in a separate folder.

**Reply by EstherPlomp** [reply]
Looks very clear - and I think it is sensible to separate data collection per instrument! 

Just a small note on using spaces in folder/directory names: This may complicate things when working in the command line or when working with scripts, in that case it is easier to name directories with - or _ instead of spaces.

**Comment by annickteepe** [comment]
I never use python, so the cookiecutter thing does not work for me. Next to this most of my data is not on my laptop. To be honest I don't really know how to start / approach this problem of making a better folder structure. Maybe someone has some suggestions or tips? Here is the explanation of my situation:

- In a git repository (shared with group members), the MATLAB software called 'NVAFM' is maintained. This software was once made in Harvard over 15 years ago by many different people and contains millions of lines of uncommented code and scripts. So in this folder, with many sub, subsub and subsubsub folders, you can find all the code to make the matlab GUIs work. I use these guis to control my instruments (RF sources, photon detectors, scanning mirrors, etc) and run measurements. But also my own measurement and analysis scripts are in this repository. The measurement/analysis scripts of groupmembers are also in here, stored in different folders with different  organisation methods. It is a very messy repository and after working with it for more than a year I still do not understand it 
- On my lab pc, all raw data, intermediate, and analyzed data is stored. See also the following included screenshot of the folder which contains all the data, including the folder to the NVAFM_matlab_software git repository. 
![folder_structure_lab_pc](https://user-images.githubusercontent.com/115211164/224734891-bd0fb59e-b0a5-46fa-81fc-74a942b59f38.png)
Most of the folders here (called e.g. track_data, pulse_data,pulse_generation, esr_data) contain raw data, automatically saved  by the matlab guis. Some folders here are made by the person who worked on the setup before me - I have no idea what kind of data is saved here, and whether I should keep it or not. In the folder Data_Annick, all the data I have gathered in my PhD is saved. This is not raw data, but intermediate or analysed data. I have organised this folder by date. 
- The full folder you see in the image (called NVAFM) is backed up to the NAS of our group.
- On my own laptop, I have other folders containing e.g. presentations, materials from courses, or teaching material. This is not well structured either but I find it hard to distinguish the types of data and sort them accordingly. This folder is synced with onedrive. 
- Finally, i have a OneDrive notebook in which I keep a detailed lab journal. I access this from both my laptop and the lab PC.

**Reply by EstherPlomp** [reply]
No worries about not using cookie cutter! You can also have a look at this discussion thread to see what folder structures other people have set up. 

And oof - this is not a one person mission I'm afraid. It is very difficult if you have multiple people working together and years of data collected in the wild. Is there a way to raise this with your supervisor and discuss a more structured way of organising this? They should have more experience with managing this - or at least know a person who would be able to help. If there are no agreed upon ways to work, it will be difficult to structure things. In research labs it is generally the PI, sometimes technicians, that can set out these rules. 

I would also discuss the folders with unknown content with your supervisor. If you're not sure about the contents and you're currently not using it, it might be an idea to request a project drive for this data and store it somewhere safely - but out of your way. 

If you appreciate a brainstorm about how to handle this, or a mediator between yourself and your supervisor regarding this - let me know!

On how to organise presentations, course materials etc: @rooswouters shared a structure about this above as well. 

I hope this provides some starting pointers - let me know if you need more support!

**Comment by MaximDeSmet** [comment]
I will request a project drive and arrange it according to this structure.

├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── bin                <- Compiled and external code, ignored by git (PG)
│   └── external       <- Any external source code, ignored by git (RO)
├── config             <- Configuration files (HW)
├── scripts           <- All relevant python scripts
│   ├── measurement 
│   └── analysis 
├── data               <- All project data, ignored by git
│   ├── processed      <- The final, canonical data sets for modeling. (PG)
│   │  ├── device ID nr1
│   │  │ └── processed data file
│   │  ├── device ID nr2
│   │  └── etc
│   ├── raw            <- The original, immutable data dump. (RO)
│   │  ├── device ID nr1
│   │  │ └── raw data file
│   │  ├── device ID nr2
│   │  └── etc
├── docs               <- Documentation users (HW)
│   ├── set-up     <-  Experimental set-up documentation and changes (HW)
│   ├── manuscript     <- Manuscript source, e.g., LaTeX, (HW)
│   └── reports        <- Other project reports and notebooks (e.g. .ppt, .Rmd) (HW)
├── results
│   ├── figures        <- Figures for the manuscript or reports (PG)
│   └── output         <- Other output for the manuscript or reports (PG)
└── src                <- Source code for this project (HW)

**Reply by EstherPlomp** [reply]
Looks very clear and sensible to divide the data by which method is used to generate it! 

Small note that using spaces in folder/directory names (device ID nr1) may complicate things when working in the command line or when working with scripts, in that case it is easier to name directories with - or _ instead of spaces. Which you might already do because I see 'set-up' and you're probably just using these names as an example, but I'll point it out to be sure :) (also it might suffice to just use the number and remove nr in front of it)

**Comment by bejuarez96** [comment]
My folder structure will look like this:

├── README  <- Contains the information about the project including the license, citation and           consortium information
├── 1 Data (.xlsx, .pdf)              <- All project data
│   ├── data collection method <- Describes the methods used to obtain the data
│   ├── data dictionary <- Describes the variables, parameters and units used
│   ├── external      <- Data from collaborators
│   ├── processed   <- The final data processed/calculated
│   ├── intermediate       <- Intermediate data, including the drafts
│   └── raw                <- Original data obtained
├── 2 Documents (.docx, .pdf)               <- Documentation 
│   ├── digital lab notebook    <- It contains a backup copy of the physical lab journal
│   ├── literature reports     <- Background information about the project
│   ├── reports     <- Research reports with analysis of the results and conclusions
│   ├── thesis    <- Final thesis
│   ├── articles        <- Articles to publish
│   └── drafts        <- Drafts of articles and reports
├── 3 Figures (.jpg)
│   ├── figures        <- Figures/schemes for the manuscript or reports

**Reply by EstherPlomp** [reply]
Looks clear! 

It might be an idea to organise your figures within the article/thesis chapter folders - so that you can find these back easily per chapter/article written. 

And is there a way to further subdivide your data? Would it make sense to set up sub-folders per data collection method?

**Reply by bejuarez96** [reply]
Yes @EstherPlomp, I was thinking the same, it would be better to put the figures in the each folder of the reports. I think I can further subdivide the data having specific folders for the microscope and FACS data. Thanks for your feedback !

**Comment by phvree** [comment]
I have some of the same problems Annick is facing, a large part of the measurement and analysis code is based on previous work and stored in a git repository, but most of the analysis and final reports are stored on a external NAS. Here, the scripts and the data do not follow the  folder structure of the cookiecutter template. Although, the folder structure should definitely improved, I also feel like it is important that I do it in such a way that mimics the folder structure of the rest of the group, for an even easier collaboration. 

The current structure of storing consist of:
- A notebook in onenote on the lab PC (synced so that we can acces is online)
- Measurement code (and basis analysis and intermediate code) in a github repository
-  Measurement data on the lab PC - backed up to a NAS folder where we access the data for further analysis , with a structure of
──./NAS/Measurement_data/Setup/
│└── type of sample
│ │ └── sample
│ ││ └── analysis                  <- scripts  for intermediate and final analysis
│ ││ └── Measurement data <- raw and intermediate data

I am not yet entirely sure how I can move al of this in a single, easily research compendium and to have consistency with the work before, but my current guess is something along the line of this.

Within the NAS environment I create/alter the folder strucutre of the project/ setup to

──./NAS/projects/Setup/ with every project on the setup getting an folder structure like this
.

├── README.md <- Contains the information about the project including the license, citation and consortium information, plus where the external data is stored as well and how to acces it
├── requirements.txt  <- short basic txt overview of how the current measurement setup looks and measurement shemes
├──data               <- All project data, ignored by git
│   ├── processed      <- The final, canonical data sets for modeling. (PG)
│   ├── raw             <-Here I store all relevant measurement data 
│   └── temp           <- Intermediate data that has been transformed. (PG)
├── docs               <- Documentation notebook for users (HW)
│   ├── Onenote    <- either a copy of the notebook or a direct link to acces the notebook
│   └── reports        <- Other project reports and notebooks
├── Analysis
│   ├── basic       <- Basic script to model and extract usefull data from the results
│   └── clean         <- scripts to visualize the data in a clean manner and call basic analysis scripts
├── results
│   ├── figures        <- Figures for the manuscript or reports (PG)
│   └── output         <- Other output for the manuscript or reports (PG)
└── src                <- Location to the github repository where the measurement code is stored

**Reply by EstherPlomp** [reply]
>  I also feel like it is important that I do it in such a way that mimics the folder structure of the rest of the group, for an even easier collaboration.

Absolutely important to follow practices of the group. Perhaps you can discuss suggestions for improvement with them. Nevertheless, in the end, whatever the majority or the PI decides might triumph all other arguments :) 

Folder structures are also not set in stone - if you find out that a different way works better you can always adjust. Important thing is to remain consistent and update the information in the readme file if you change practices. I think what you came up with here looks great!

**Comment by rwmulder** [comment]
I was able to clone the cookiecutter template but didn't really know what to do next so I edited the folder structure manually. This is the folder structure I would ideally like to use. However, I'm collaborating with others and there is already some structure in place. I'm not a huge fan of our current structure but converting the whole thing would cost a lot of effort since there is a lot in there from before my time.

├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── Project Management
│	├── Meeting Notes
│	├── Plannings
│	├── Presentations
│   └── Financial 
├── Data
│	├── Lab logbook
│   ├── Cleanroom
│   ├── Transport
│   └── NV magnetometry
├── Code
│	├── Analysis
│	├── Simulation
│	├── Measurement Scripts
│	└── Misc
├── Instrumentation         
│   ├── Manuals    
│   └── Notes    
├── Writing         
│   ├── Manuscript     
│   ├── Figures        
│   └── Output         
├── Theory              
│   ├── Literature
│   └── Notes 
├── Teaching
│	├── Supervision
│	└── TA
├── Graduate School
└── Misc

**Reply by EstherPlomp** [reply]
Not surprisingly I see similar struggles as @phvree and @annickteepe described, so I'll also refer you to my comments/feedback below their comments. 

I think your folder structure looks very clear and structured! Only a small note about spaces in folder names - if you use those in scripts the terminal/scripts may have difficulties reading folder names, so using separators like - and _ is generally safer. That is less urgent in folder names such as "meeting notes".

**Comment by cfchen98** [comment]
My data folders are organized in this way:
├── Project name
README <- General info
├── 1 Raw data (.mpr) <- All project data
README <- data collection method
│ ├── external <- Data from collaborators
│ ├── README <- data collection method
│ ├── Cyclic voltammetry
│ ├── Galvanostatic charge and discharge
│ ├── Impedence
├── 2 Processed data (.opju,.xsl) 
README <- data processing method
│ ├── Cyclic voltammetry
│ ├── Galvanostatic charge and discharge
│ ├── Impedence
│ ├── external <- Data from collaborators
├── 3 Figures (.Tiff)
│ ├── figures <- Figures/schemes for the manuscript or reports
├── 4 Manuscript 
│ ├── articles <- Articles to publish
│ └── drafts <- Drafts of articles and reports

**Reply by EstherPlomp** [reply]
Looks very clear and structured! 

Just so that I understand correctly: You'll have a readme file for the project folder, in the raw data folder and in the processed data folder? 

And will you have separate folders for each of the articles or chapters that you'll work on?

**Reply by cfchen98** [reply]
Thanks for your feedback. Yes! That 's my plan to have a readme file in each folder. Since there are different types of data being collected in different layer in the folder, so it is better to build up a detailed description for each layer. However, I am still stuggling creating a general template for each session.

**Reply by EstherPlomp** [reply]
Thanks for the clarification: that makes sense! 

Where are you struggling? Are you not sure what information needs to be there? Would it help to start with the information that needs to be there for each stage?

**Comment by Azinkarimzad** [comment]
My data structure would be like this:
.
├── .gitignore
├── CITATION.md
├── LICENSE.md
├── README.md
├── requirements.txt
├── src                <- Compiled and external code, ignored by git. 
│   └── external_delph <- Any external source code from delphi, ignored by git.
│   └── external_else  <- Any external source code from other sources, ignored by git. 
├── Geo_data           <- All project data, ignored by git.
│   ├── processed      <- The final, data sets for modeling. 
│   ├── raw            <- The original, immutable data dump. 
│   └── temp           <- Intermediate data that has been transformed. 
├── Seis_data          <- All project data, ignored by git.
│   ├── processed      <- The final, data sets for modeling. 
│   ├── raw            <- The original, immutable data dump. 
│   └── temp           <- Intermediate data that has been transformed. 
├── ML_data            <- All project data, ignored by git.
│   ├── input          <- The initial,  data sets for ML modeling. 
│   ├── output         <- The final,  ML sets for ML modeling. 
├── Scripts            <- Documentation notebook for users. 
│   ├── Python         <- Script source (geo and ML), e.g., python. 
│   └── Matlab         <- Script source (geo and ML). 
├── docs               <- Documentation notebook for users.
│   ├── manuscript     <- Manuscript source, e.g., LaTeX, Word, ppt, etc. 
│   └── reports        <- Other project reports and notebooks (e.g. Jupyter) 
├── results
│   ├── figures        <- Figures for the manuscript or reports.
│   ├── developed      <- Methods, scripts that are mainly developed within the project.
│   └── output         <- Other output for the manuscript or reports.
├── MSC                <- Supervision activities. 
│   └── Msc_doc        <- Any document related to supervision.
│   └── Msc_script     <- Any scripts related to supervision.
├── courses            <- Courses taken during PhD. 
│   └── CS             <- Courses_ material.
│   └── CS_CER         <- Course certificate.
├── Emails             <- Important emails copy.

**Reply by EstherPlomp** [reply]
Looks very clear @Azinkarimzad! Well done!

Will you have seperate directories/folders for each of your manuscript chapters/articles?

**Reply by Azinkarimzad** [reply]
@EstherPlomp 
Indeed that's a god idea. I will include that as well.

**Comment by yzhang0429** [comment]
I wasn't able to use cookiecutter on our group drive where we have our data stored, so I manually organized the folder structure:
│ │ └── Wafer number
│ │ │ └── chip number <- main folder storing all the information on this chip
│ │ │ │ └── design <- CAD files for the lithography
│ │ │ │ └── image <- all the inspection image taken during the fabrication
│ │ │ │  │ └── 0. fresh  <- each step has a separated folder
│ │ │ │  │ └── 1. Mesa etch 
│ │ │ │  └── ```
│ │ │ │ └── data <- measurement script and database file
│ │ │ │ └── analysis <- analysis script and all the generated database/ plot/ etc.

This is an established folder structure shared by all the project team.

**Reply by EstherPlomp** [reply]
No worries about not using cookiecutter for the assignment! 

When using code/scripts - be careful of using spaces in folder names; this is not machine readable so you might run into issues if you use spaces (- and _ are better alternatives).

**Reply by yzhang0429** [reply]
That's true! Usually we don't have to use code for the image folder, but it's nice to have consistent naming with '-' or '_'. Thanks for the feedback!

**Comment by arr0w-hs** [comment]
<img width="768" alt="Screenshot 2023-03-28 at 11 11 58" src="https://user-images.githubusercontent.com/28155151/228188492-9d1b9a18-6562-4072-8023-51ff9ed4a6fa.png">

This is what my folder structure looks so far in the project. I have different projects/directions that i could take as different folders. All the folders are divided based on the time into further folders. I will also incorporate the cookiecutter template as my folders get more populated. I also have been tagging different files based on my progress.

**Reply by EstherPlomp** [reply]
Many thanks for sharing the folder structure! Do I understand it correctly that I'm primarily looking at literature right now? 
Tagging is a very helpful feature!
